src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,441 @@
+
+    response< int, std::map<std::string, std::string>,
+	      boost::redis::ignore_t > resp;
+
+    conn->async_exec(req, resp, yield[ec]);
+
+    ASSERT_EQ((bool)ec, false);
+    EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+    EXPECT_EQ(std::get<1>(resp).value().size(), 10);
+
+    conn->cancel();
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.run();
+}
+
+TEST_F(D4NReadCacheFixture, DeleteObject)
+{
+  const std::string testName = "DeleteObject";
+ 
+  boost::asio::spawn(io, [this, testName] (boost::asio::yield_context yield) {
+    init_driver(this->driver, yield);
+    create_bucket(testName, this->driver, &this->testBucket, yield);
+    put_object(testName, &this->obj, this->driver, &this->testBucket, &this->testWriter, yield);
+
+    std::unique_ptr<rgw::sal::Object::ReadOp> read_op(obj->get_read_op());
+    ASSERT_EQ(read_op->prepare(optional_yield{yield}, env->dpp), 0);
+    ASSERT_EQ(read_op->iterate(env->dpp, 0, ofs, nullptr, optional_yield{yield}), 0);
+
+    std::error_code err;
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/test_version", err), true);  
+    std::string oid = "test_version#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), true);     
+
+    {
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0"); // TODO: add null head object check as well, or is that being removed?
+      req.push("FLUSHALL");
+
+      response< int, boost::redis::ignore_t > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+    }
+
+    std::unique_ptr<rgw::sal::Object::DeleteOp> del_op = obj->get_delete_op();
+    EXPECT_EQ(del_op->delete_obj(env->dpp, optional_yield{yield}, rgw::sal::FLAG_LOG_OP), 0);
+
+    /* TODO: Eviction cycle to delete cache blocks
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/test_version", err), false);  
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), false);     
+    */
+
+    {
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0"); // TODO: add null head object check as well, or is that being removed?
+      req.push("FLUSHALL");
+
+      response< int, boost::redis::ignore_t > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 0);
+    }
+
+    conn->cancel();
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.run();
+}
+
+TEST_F(D4NWriteCacheFixture, PutObject)
```

Comment:
why is a separate fixture needed for a Write cache?

Reply:
The initialization is different since I must change the config value but I can change the value at the beginning of each test instead of creating a separate fixture entirely if you believe that is the cleaner approach

Reply:
If it is easier for you to get the tests done with separate fixtures, then go for it. We can combine them later when we have the time.